src/dao/AbstractDAO.ts

```diff
@@ -154,19 +178,24 @@ export abstract class AbstractDAO<K, T> {
    * Create an [[OnmsHTTPOptions]] object for DAO calls given an optional filter.
    * @param filter - the filter to use
    */
-  protected getOptions(filter?: Filter): OnmsHTTPOptions {
-    const ret = new OnmsHTTPOptions();
-    if (this.useJson()) {
-      ret.headers.accept = 'application/json';
-    } else {
-      // always use application/xml in DAO calls when we're not sure how
-      // usable JSON output will be.
-      ret.headers.accept = 'application/xml';
-    }
-    if (filter) {
-      ret.parameters = this.filterProcessor.getParameters(filter);
-    }
-    return ret;
+  protected async getOptions(filter?: Filter): Promise<OnmsHTTPOptions> {
+      return Promise.resolve(new OnmsHTTPOptions())
+          .then((options) => {
+              if (this.useJson()) {
+                  options.headers.accept = 'application/json';
+              } else {
+                  // always use application/xml in DAO calls when we're not sure how
+                  // usable JSON output will be.
+                  options.headers.accept = 'application/xml';
+              }
+              if (filter) {
+                  return this.getFilterProcessor().then((processor) => {
```

Comment:
You could avoid having to refactor all of the `getOptions` code into async by using `await` here and making it synchronous again.